pallets/circuit/src/lib.rs

```diff
@@ -1325,43 +1353,45 @@ impl<T: Config> Pallet<T> {
 
         match local_ctx.xtx.status {
             CircuitStatus::RevertTimedOut | CircuitStatus::Reverted => {
-                for fse in &local_ctx.full_side_effects[(current_step) as usize] {
```

Comment:
Oh I see, this was the buggy part, nice catch

Reply:
@AwesomeIbex what's the context behind this change here?

Reply:
@MaciejBaj if you try to index something that doesnt have any side effects the circuit will panic loop.